platform/ports/posix/types/iot_platform_types_posix.h

```diff
@@ -32,6 +32,9 @@
     #include POSIX_TYPES_HEADER
 #else
     #include <sys/types.h>
+    #include <arpa/inet.h>
```

Comment:
This won't work for the mac OS port, which doesn't include this file.

Can you move these to iot_network_mbedtls.c?
```c
#if !defined( _WIN32 ) && !defined( _WIN64 )
    #include <arpa/inet.h>
    #include <netdb.h>
    #include <sys/socket.h>
#endif
```